### PR TITLE
K3s Helper: Ensure each k3s channel applies to one version.

### LINKED ISSUE #1584: Outdated k8s versions
### Actual Behavior

My k8s list shows an outdated set of versions, compared both to a colleague and to @jandubois.

Mine:
image

Colleague:
image

Jan:
image

### Steps to Reproduce

Start RD, view Preferences > Kubernetes Settings

### Result

Different available versions from others

### Expected Behavior

Same available versions

### Additional Information

_No response_

### Rancher Desktop Version

1.0.1

### Rancher Desktop K8s Version

1.21.7

### Which container runtime are you using?

containerd (nerdctl)

### What operating system are you using?

Windows

### Operating System / Build Version

Windows 10 Enterprise, Version 21H2 (OS Build 19044.1526)

### What CPU architecture are you using?

x64

### Linux only: what package format did you use to install Rancher Desktop?

_No response_

### Windows User Only

Nothing relevant

### PR DESCRIPTION
**Pull request by mook-as**
As the channel data changes, we can end up with the same k3s channel label being applied to multiple releases (due to restoring data from the cache). Change the cache format & the updating code to prevent this from happening.

Fixes #1584.

### CODE REVIEW

**Review by ericpromislow — CHANGES_REQUESTED** [review]
Code looks right, but I found a problem:
Initially, 1.22.6 was marked as the stable version.
I shut down the app, and manually edited the cache file to make `1.22.4` stable.
Restarted the app.
1.22.4 is stable, but I expected it to bounce back to 1.22.6 (or a newer version, if one arrived while I was doing this.

**Comment by ericpromislow** [comment]
Also, this isn't backwards-compatible. When I rerun main on the newer cache:
```
2022-03-01T21:57:10.892Z: k3sHelper.initialize failed:  TypeError: cacheData is not iterable
    at K3sHelper.readCache (/Users/ericp/workspace/rancher/desktop/dist/app/background.js:36349:33)
    at async /Users/ericp/workspace/rancher/desktop/dist/app/background.js:36534:17
```

Not sure this is actionable, and it's a low-risk problem, but it's a good reason to always put a version in a state file.

**Comment by ericpromislow** [comment]
I think it's a mistake to ever throw an exception from trying to read from the cache. If there's a problem, just return (maybe ensuring that `this.versions` is cleared) and rebuild the cache file.

**Comment by mook-as** [comment]
- Fixed the issue of keeping stale tags on the wrong version.
- Can't do anything about older versions, but fixed the _current_ code (even though it has a version tag now) to always ignore errors reading the cache.

File: src/k8s-engine/k3sHelper.ts [hunk 1]
```diff
@@ -41,6 +41,58 @@ export interface ReleaseAPIEntry {
   }[];
 }
 
+const CURRENT_CACHE_VERSION = 2 as const;
```
Comment (ericpromislow) [inline src/k8s-engine/k3sHelper.ts:44]: What does `as const` mean? Is that a typo? Is it even needed?

Comment (mook-as) [reply src/k8s-engine/k3sHelper.ts:44]: It's TypeScript weirdness.
`const CURRENT_CACHE_VERSION = 2` means TypeScript deduces `CURRENT_CACHE_VERSION` to have a type of `number` (that is, _any_ number).
`const CURRENT_CACHE_VERSION = 2 as const` means "no really, just this one value", i.e. the type is `2` exactly, and no other number.

**Review by ericpromislow — APPROVED** [review]
LGTM

**Pull request merged**